### ISSUE TITLE
[BUG] AttributeError: 'nvstrings' object has no attribute 'to_gpu_array'

### ISSUE THREAD

**Issue by Cdivyam** [rapidsai/cudf#2169]
I am trying to use the `apply_rows` function of cudf but getting the above error:

Input:
```
train_word_match_cudf = cudf_train.apply_rows(word_match_share_cudf, incols = ['question1', 'question2'], outcols = dict(R = np.float64), kwargs = dict())
```
word_match_share_cudf function:
```
from nltk.corpus import stopwords

stops = set(stopwords.words("english"))

def word_match_share_cudf(q1, q2, R): #important function
    q1words = {}
    q2words = {}
    for word in str(q1).lower().split():
        if word not in stops:
            q1words[word] = 1
    for word in str(q2).lower().split():
        if word not in stops:
            q2words[word] = 1
    if len(q1words) == 0 or len(q2words) == 0:
        # The computer-generated chaff includes a few questions that are nothing but stopwords
        return 0
    shared_words_in_q1 = [w for w in q1words.keys() if w in q2words]
    shared_words_in_q2 = [w for w in q2words.keys() if w in q1words]
    R = (len(shared_words_in_q1) + len(shared_words_in_q2))/(len(q1words) + len(q2words))
    return R
```
The cudf_train dataframe consists of:
![image](https://user-images.githubusercontent.com/39796302/60583403-72f16500-9da8-11e9-984a-c97209af59fd.png)

**Expected behavior**
The result should be a Series named `train_word_match_cudf` where each row is a value of datatype float.

**Environment Overview**
 - Environment location: Docker
 - Method of cuDF install: docker
`docker pull nvcr.io/nvidia/rapidsai/rapidsai:cuda9.2-runtime-ubuntu16.04`

**Environment details**
```
**git***
     Not inside a git repository

     ***OS Information***
     DISTRIB_ID=Ubuntu
     DISTRIB_RELEASE=16.04
     DISTRIB_CODENAME=xenial
     DISTRIB_DESCRIPTION="Ubuntu 16.04.6 LTS"
     NAME="Ubuntu"
     VERSION="16.04.6 LTS (Xenial Xerus)"
     ID=ubuntu
     ID_LIKE=debian
     PRETTY_NAME="Ubuntu 16.04.6 LTS"
     VERSION_ID="16.04"
     HOME_URL="http://www.ubuntu.com/"
     SUPPORT_URL="http://help.ubuntu.com/"
     BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
     VERSION_CODENAME=xenial
     UBUNTU_CODENAME=xenial
     Linux 6d06c642df9e 4.15.0-45-generic #48-Ubuntu SMP Tue Jan 29 16:28:13 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

     ***GPU Information***
     Wed Jul  3 12:10:43 2019
     +-----------------------------------------------------------------------------+
     | NVIDIA-SMI 410.104      Driver Version: 410.104      CUDA Version: 10.0     |
     |-------------------------------+----------------------+----------------------+
     | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
     | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
     |===============================+======================+======================|
     |   0  Tesla V100-SXM2...  On   | 00000000:06:00.0 Off |                    0 |
     | N/A   39C    P0    56W / 300W |    478MiB / 16130MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   1  Tesla V100-SXM2...  On   | 00000000:07:00.0 Off |                    0 |
     | N/A   40C    P0    47W / 300W |     11MiB / 16130MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+

     +-----------------------------------------------------------------------------+
     | Processes:                                                       GPU Memory |
     |  GPU       PID   Type   Process name                             Usage      |
     |=============================================================================|
     +-----------------------------------------------------------------------------+

     ***CPU***
     Architecture:          x86_64
     CPU op-mode(s):        32-bit, 64-bit
     Byte Order:            Little Endian
     CPU(s):                80
     On-line CPU(s) list:   0-79
     Thread(s) per core:    2
     Core(s) per socket:    20
     Socket(s):             2
     NUMA node(s):          2
     Vendor ID:             GenuineIntel
     CPU family:            6
     Model:                 79
     Model name:            Intel(R) Xeon(R) CPU E5-2698 v4 @ 2.20GHz
     Stepping:              1
     CPU MHz:               2346.903
     CPU max MHz:           3600.0000
     CPU min MHz:           1200.0000
     BogoMIPS:              4391.84
     Virtualization:        VT-x
     Hypervisor vendor:     vertical
     Virtualization type:   full
     L1d cache:             32K
     L1i cache:             32K
     L2 cache:              256K
     L3 cache:              51200K
     NUMA node0 CPU(s):     0-19,40-59
     NUMA node1 CPU(s):     20-39,60-79
     Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsrsse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single pti intel_ppin ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm rdt_a rdseed adx smap intel_pt xsaveopt cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts flush_l1d

     ***CMake***
     /conda/envs/rapids/bin/cmake
     cmake version 3.12.4

     CMake suite maintained and supported by Kitware (kitware.com/cmake).

     ***g++***
     /usr/bin/g++
     g++ (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
     Copyright (C) 2015 Free Software Foundation, Inc.
     This is free software; see the source for copying conditions.  There is NO
     warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


     ***nvcc***
     /usr/local/cuda/bin/nvcc
     nvcc: NVIDIA (R) Cuda compiler driver
     Copyright (c) 2005-2018 NVIDIA Corporation
     Built on Tue_Jun_12_23:07:04_CDT_2018
     Cuda compilation tools, release 9.2, V9.2.148

     ***Python***
     /conda/envs/rapids/bin/python
     Python 3.6.8

     ***Environment Variables***

     PATH                            : /conda/envs/rapids/bin:/conda/envs/rapids/bin:/conda/condabin:/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/conda/bin
     LD_LIBRARY_PATH                 : /usr/local/nvidia/lib:/usr/local/nvidia/lib64:/usr/local/cuda/lib64
     NUMBAPRO_NVVM                   : /usr/local/cuda/nvvm/lib64/libnvvm.so
     NUMBAPRO_LIBDEVICE              : /usr/local/cuda/nvvm/libdevice
     CONDA_PREFIX                    : /conda/envs/rapids
     PYTHON_PATH                     :

     ***conda packages***
     /conda/condabin/conda
     # packages in environment at /conda/envs/rapids:
     #
     # Name                    Version                   Build  Channel
     arrow-cpp                 0.12.1           py36h0e61e49_0    conda-forge
     atomicwrites              1.3.0                      py_0    conda-forge
     attrs                     19.1.0                     py_0    conda-forge
     backcall                  0.1.0                      py_0    conda-forge
     bleach                    3.1.0                      py_0    conda-forge
     bokeh                     1.1.0                    py36_0    conda-forge
     boost                     1.68.0          py36h8619c78_1001    conda-forge
     boost-cpp                 1.68.0            h11c811c_1000    conda-forge
     bzip2                     1.0.6             h14c3975_1002    conda-forge
     ca-certificates           2019.3.9             hecc5488_0    conda-forge
     certifi                   2019.3.9                 py36_0    conda-forge
     cffi                      1.12.3           py36h8022711_0    conda-forge
     click                     7.0                      pypi_0    pypi
     cloudpickle               1.0.0                      py_0    conda-forge
     cmake                     3.12.4            h8d4ced6_1000    conda-forge
     cudatoolkit               9.2                           0
     cudf                      0.7.2                    py36_0    rapidsai/label/cuda9.2
     cugraph                   0.7.0                    py36_0    rapidsai/label/cuda9.2
     cuml                      0.7.0            cuda9.2_py36_0    rapidsai/label/cuda9.2
     curl                      7.64.1               hf8cf82a_0    conda-forge
     cycler                    0.10.0                     py_1    conda-forge
     cython                    0.29.7           py36he1b5a44_0    conda-forge
     cytoolz                   0.9.0.1         py36h14c3975_1001    conda-forge
     dask                      1.2.2                      py_0    conda-forge
     dask-core                 1.2.2                      py_0    conda-forge
     dask-cuda                 0.7.0                    py36_0    rapidsai/label/cuda9.2
     dask-cudf                 0.7.0                    py36_0    rapidsai/label/cuda9.2
     dask-cuml                 0.7.0                    py36_0    rapidsai/label/cuda9.2
     dask-xgboost              0.1.5                    pypi_0    pypi
     dbus                      1.13.6               he372182_0    conda-forge
     decorator                 4.4.0                      py_0    conda-forge
     defusedxml                0.5.0                      py_1    conda-forge
     distributed               1.28.1                   py36_0    conda-forge
     entrypoints               0.3                   py36_1000    conda-forge
     expat                     2.2.5             hf484d3e_1002    conda-forge
     fontconfig                2.13.1            he4413a7_1000    conda-forge
     freetype                  2.10.0               he983fc9_0    conda-forge
     gettext                   0.19.8.1          hc5be6a0_1002    conda-forge
     glib                      2.58.3            hf63aee3_1001    conda-forge
     gst-plugins-base          1.14.4            hdf3bae2_1001    conda-forge
     gstreamer                 1.14.4            h66beb1c_1001    conda-forge
     heapdict                  1.0.0                 py36_1000    conda-forge
     icu                       58.2              hf484d3e_1000    conda-forge
     ipykernel                 5.1.1            py36h24bf2e0_0    conda-forge
     ipython                   7.3.0            py36h24bf2e0_0    conda-forge
     ipython_genutils          0.2.0                      py_1    conda-forge
     jedi                      0.13.3                   py36_0    conda-forge
     jinja2                    2.10.1                     py_0    conda-forge
     joblib                    0.13.2                     py_0    conda-forge
     jpeg                      9c                h14c3975_1001    conda-forge
     jsonschema                3.0.1                    py36_0    conda-forge
     jupyter_client            5.2.4                      py_3    conda-forge
     jupyter_core              4.4.0                      py_0    conda-forge
     jupyterlab                0.35.6                   py36_0    conda-forge
     jupyterlab_server         0.2.0                      py_0    conda-forge
     kiwisolver                1.1.0            py36hc9558a2_0    conda-forge
     krb5                      1.16.3            h05b26f9_1001    conda-forge
     libblas                   3.8.0                9_openblas    conda-forge
     libcblas                  3.8.0                9_openblas    conda-forge
     libcudf                   0.7.2                 cuda9.2_0    rapidsai/label/cuda9.2
     libcugraph                0.7.0                 cuda9.2_0    rapidsai/label/cuda9.2
     libcuml                   0.7.0                 cuda9.2_0    rapidsai/label/cuda9.2
     libcurl                   7.64.1               hda55be3_0    conda-forge
     libedit                   3.1.20181209         hc058e9b_0
     libffi                    3.2.1                hd88cf55_4
     libgcc-ng                 8.2.0                hdf63c60_1
     libgfortran-ng            7.3.0                hdf63c60_0
     libiconv                  1.15              h516909a_1005    conda-forge
     liblapack                 3.8.0                9_openblas    conda-forge
     libnvstrings              0.7.0                 cuda9.2_0    rapidsai/label/cuda9.2
     libpng                    1.6.37               hed695b0_0    conda-forge
     libprotobuf               3.6.1             hdbcaa40_1001    conda-forge
     librmm                    0.7.0                 cuda9.2_0    rapidsai/label/cuda9.2
     libsodium                 1.0.16            h14c3975_1001    conda-forge
     libssh2                   1.8.2                h22169c7_2    conda-forge
     libstdcxx-ng              8.2.0                hdf63c60_1
     libtiff                   4.0.10            h648cc4a_1001    conda-forge
     libuuid                   2.32.1            h14c3975_1000    conda-forge
     libuv                     1.29.0               h516909a_0    conda-forge
     libxcb                    1.13              h14c3975_1002    conda-forge
     libxml2                   2.9.9                h13577e0_0    conda-forge
     llvmlite                  0.27.0dev0      py36hf484d3e_19    numba
     locket                    0.2.0                      py_2    conda-forge
     markupsafe                1.1.1            py36h14c3975_0    conda-forge
     matplotlib                3.0.3                    py36_1    conda-forge
     matplotlib-base           3.0.3            py36h5f35d83_1    conda-forge
     mistune                   0.8.4           py36h14c3975_1000    conda-forge
     more-itertools            4.3.0                 py36_1000    conda-forge
     msgpack-python            0.6.1            py36h6bb024c_0    conda-forge
     nbconvert                 5.5.0                      py_0    conda-forge
     nbformat                  4.4.0                      py_1    conda-forge
     ncurses                   6.1                  he6710b0_1
     networkx                  2.3                        py_0    conda-forge
     notebook                  5.7.8                    py36_0    conda-forge
     numba                     0.41.0          py36h637b7d7_1000    conda-forge
     numpy                     1.16.2           py36h8b7e671_1    conda-forge
     nvstrings                 0.7.0                    py36_0    rapidsai/label/cuda9.2
     olefile                   0.46                       py_0    conda-forge
     openblas                  0.3.6                h6e990d7_2    conda-forge
     openssl                   1.1.1b               h14c3975_1    conda-forge
     packaging                 19.0                       py_0    conda-forge
     pandas                    0.23.4          py36h637b7d7_1000    conda-forge
     pandoc                    2.7.2                         0    conda-forge
     pandocfilters             1.4.2                      py_1    conda-forge
     parquet-cpp               1.5.1                         4    conda-forge
     parso                     0.4.0                      py_0    conda-forge
     partd                     0.3.9                      py_0    conda-forge
     patsy                     0.5.1                      py_0    conda-forge
     pcre                      8.41              hf484d3e_1003    conda-forge
     pexpect                   4.7.0                    py36_0    conda-forge
     pickleshare               0.7.5                 py36_1000    conda-forge
     pillow                    6.0.0            py36he7afcd5_0    conda-forge
     pip                       19.1.1                   py36_0
     pluggy                    0.11.0                     py_0    conda-forge
     prometheus_client         0.6.0                      py_0    conda-forge
     prompt_toolkit            2.0.9                      py_0    conda-forge
     psutil                    5.6.2            py36h516909a_0    conda-forge
     pthread-stubs             0.4               h14c3975_1001    conda-forge
     ptyprocess                0.6.0                   py_1001    conda-forge
     py                        1.8.0                      py_0    conda-forge
     pyarrow                   0.12.1           py36hbbcf98d_0    conda-forge
     pycparser                 2.19                     py36_1    conda-forge
     pygments                  2.4.0                      py_0    conda-forge
     pyparsing                 2.4.0                      py_0    conda-forge
     pyqt                      5.9.2            py36hcca6a23_0    conda-forge
     pyrsistent                0.15.2           py36h516909a_0    conda-forge
     pytest                    4.5.0                    py36_0    conda-forge
     python                    3.6.8                h0371630_0
     python-dateutil           2.8.0                      py_0    conda-forge
     pytz                      2019.1                     py_0    conda-forge
     pyyaml                    5.1              py36h14c3975_0    conda-forge
     pyzmq                     18.0.1           py36hc4ba49a_1    conda-forge
     qt                        5.9.7                h52cfd70_1    conda-forge
     readline                  7.0                  h7b6447c_5
     rhash                     1.3.6             h14c3975_1001    conda-forge
     rmm                       0.7.0                    py36_0    rapidsai/label/cuda9.2
     scapy                     2.4.2                    pypi_0    pypi
     scikit-learn              0.21.1           py36h627018c_0    conda-forge
     scipy                     1.2.1            py36h09a28d5_1    conda-forge
     seaborn                   0.9.0                      py_1    conda-forge
     send2trash                1.5.0                      py_0    conda-forge
     setuptools                41.0.1                   py36_0
     sip                       4.19.8          py36hf484d3e_1000    conda-forge
     six                       1.12.0                py36_1000    conda-forge
     sortedcontainers          2.1.0                      py_0    conda-forge
     sqlite                    3.28.0               h7b6447c_0
     statsmodels               0.9.0           py36h3010b51_1000    conda-forge
     tblib                     1.4.0                    pypi_0    pypi
     terminado                 0.8.2                    py36_0    conda-forge
     testpath                  0.4.2                   py_1001    conda-forge
     thrift-cpp                0.12.0            h0a07b25_1002    conda-forge
     tk                        8.6.9             h84994c4_1001    conda-forge
     toolz                     0.9.0                    pypi_0    pypi
     tornado                   6.0.2            py36h516909a_0    conda-forge
     traitlets                 4.3.2                 py36_1000    conda-forge
     wcwidth                   0.1.7                      py_1    conda-forge
     webencodings              0.5.1                      py_1    conda-forge
     wheel                     0.33.4                   py36_0
     xgboost                   0.83.dev0                pypi_0    pypi
     xorg-libxau               1.0.9                h14c3975_0    conda-forge
     xorg-libxdmcp             1.1.3                h516909a_0    conda-forge
     xz                        5.2.4                h14c3975_4
     yaml                      0.1.7             h14c3975_1001    conda-forge
     zeromq                    4.3.1             hf484d3e_1000    conda-forge
     zict                      0.1.4                    pypi_0    pypi
     zlib                      1.2.11               h7b6447c_3
```

**Comment by kkraus14** [comment]
@Cdivyam unfortunately string columns aren't yet supported for UDFs via `apply_rows`, `applymap`, or `query`. We're working on a solution here but we're not quite there yet.

**Comment by Cdivyam** [comment]
@kkraus14 
 Can this function work as a work around?
What does this function do?
Can't find any documentation for it.
```
from librmm_cffi import librmm
import nvstrings

def on_gpu(words,func,dtype=np.int32):
    res = librmm.device_array(words.size(), dtype=dtype)
    cmd = 'words.%s(res.device_ctypes_pointer.value)'%(func)
    eval(cmd)
    return res
```

**Comment by drabastomek** [comment]
Hit the same when trying to .query(...) on a nvstring column. Just to bump this up.

**Comment by surjya24** [comment]
Have the same issue with apply_rows. Bumping this up.

**Comment by huangyz0918** [comment]
Hit the same when trying to .query(...) on a nvstring column. Just to bump this up.

**Comment by rfox12** [comment]
does apply_chunks() have the same restriction--no strings as input?

**Comment by kkraus14** [comment]
Resolved as of 0.14 where it errors much more clearly that Strings are not yet supported for UDFs / `to_gpu_array`.
